observ/proxy.py

```diff
@@ -92,15 +101,15 @@ def to_raw(target):
         return to_raw(target.target)
 
     if isinstance(target, list):
-        return [to_raw(t) for t in target]
+        return cast(T, [to_raw(t) for t in target])
 
     if isinstance(target, dict):
-        return {key: to_raw(value) for key, value in target.items()}
+        return cast(T, {key: to_raw(value) for key, value in target.items()})
 
     if isinstance(target, tuple):
-        return tuple(to_raw(t) for t in target)
+        return cast(T, tuple(to_raw(t) for t in target))
 
     if isinstance(target, set):
-        return {to_raw(t) for t in target}
+        return cast(T, {to_raw(t) for t in target})
```

Comment:
are these `casts` in `to_raw` really necessary? `return {}` is already a plain dict right?

Reply:
@ju1ius can you share your thoughts on this one?